controller/items.go

```diff
@@ -6,13 +6,41 @@ import (
 	"net/http"
 	"strconv"
 
-	"github.com/gorilla/mux"
 	"github.com/rodrigomkd/go-rest-api/service"
+
+	"github.com/gorilla/mux"
 )
 
-func GetItems(w http.ResponseWriter, r *http.Request) {
+type IController interface {
+	GetItems(w http.ResponseWriter, req *http.Request)
+	GetItem(w http.ResponseWriter, req *http.Request)
+	GetItemsSync(w http.ResponseWriter, req *http.Request)
+}
+
+type Controller struct {
+	s service.Service
+}
+
+func New(s service.Service) *Controller {
+	return &Controller{
+		s: s,
+	}
+}
+
+func (c Controller) GetItems(w http.ResponseWriter, r *http.Request) {
+	w.Header().Set("Content-Type", "application/json")
+	items, err := c.s.GetItems()
+	if err != nil {
+		errorResponse(w, "Some Error Occurred", 500)
```

Comment:
Also, not because you received an error it means that it has to be a 500, if you didn't receive any items but everything is fine, then a 404 would be more appropriate